demisto_sdk/commands/pre_commit/hooks/sourcery.py

```diff
@@ -1,15 +1,48 @@
 import tempfile
+from copy import deepcopy
 from pathlib import Path
+from typing import Any, Dict
 
 from demisto_sdk.commands.common import tools
 from demisto_sdk.commands.pre_commit.hooks.hook import Hook
 
 
 class SourceryHook(Hook):
-    def prepare_hook(self, python_version: str, config_file_path: Path, **kwargs):
+    def _get_temp_config_file(self, config_file_path: Path, python_version: str):
```

Comment:
Basically we can.
It means that when we construct this map we need to map `Version` to python files, and modify every hook that uses python version to take `Version` instead of `str`.